share/functions/__fish_print_hostnames.fish

```diff
@@ -62,10 +62,11 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
                 | string trim | string replace -r -a '\s+' ' ')
             end
 
+            cd $relative_path
```

Comment:
Could you make it work without changing directory at all? The main reason I dislike this particularly for a commonly used completion function like __fish_print_hostnames is it could do unpleasant things to the interactive session's state, imagine being cd'd into a directory that was moved, replaced, or removed after you got there. You may not get back to where you were before hitting tab.